src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_INDEX = 2;
+    private int capacity;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        table = (Node<K, V>[]) new Node[capacity];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int index = getBucketIndex(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else if (table[index].key == newNode.key || table[index].key != null
+                && table[index].key.equals(newNode.key)) {
+            table[index].value = newNode.value;
+            return;
+        } else {
+            Node<K, V> node = table[index];
+            while (node.next != null) {
+                node = node.next;
+                if (node.key == newNode.key || node.key != null && node.key.equals(newNode.key)) {
+                    node.value = newNode.value;
+                    return;
+                }
+            }
+            node.next = newNode;
+        }
+        size++;
+        if (size > threshold) {
+            resize();
```

Comment:
Maybe you can move it at the top of method. But I think you idea is valid too

Reply:
Yes, it's possible) But I prefer my variant, if you don't mind